homeassistant/components/openweathermap/coordinator.py

```diff
@@ -192,12 +207,13 @@ def _calc_precipitation_kind(rain, snow):
     @staticmethod
     def _get_precipitation_value(precipitation):
         """Get precipitation value from weather data."""
-        if "all" in precipitation:
-            return round(precipitation["all"], 2)
-        if "3h" in precipitation:
-            return round(precipitation["3h"], 2)
-        if "1h" in precipitation:
-            return round(precipitation["1h"], 2)
+        if precipitation is not None:
```

Comment:
When would it be none?

Reply:
The pyopenweathermap library >0.2.0 seems to have introduced the possibility of None values being included for rain and snow: `wind_gust=14.92, wind_bearing=140, rain=None, snow=None`

I'll raise, and possibly fix, the issue on pyopenweathermap`.